flink-streaming-java/src/main/java/org/apache/flink/streaming/api/TimeCharacteristic.java

```diff
@@ -19,12 +19,26 @@
 package org.apache.flink.streaming.api;
 
 import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.api.common.ExecutionConfig;
+import org.apache.flink.api.common.eventtime.WatermarkStrategy;
 
 /**
- * The time characteristic defines how the system determines time for time-dependent
- * order and operations that depend on time (such as time windows).
+ * The time characteristic defines how the system determines time for time-dependent order and
+ * operations that depend on time (such as time windows).
+ *
+ * @deprecated In Flink 1.12 the default stream time characteristic has been changed to {@link
```

Comment:
nit: I am not entirely sure, but I think 1.12 is the first release with python DataStream API. If that's the case I think we don't need to mention the change of the default value here.

It doesn't harm though, therefore I am fine with either.

Reply:
Why do you mention Python here?

Reply:
Sorry, my bad. Before this class there were python files and I somehow confused this with the python equivalent.

BTW, do you think we should deprecate the `setStreamTimeCharacteristic` in python as well (or even remove it, as it was added in 1.12?)